File: TesteAndroidSantander/app/src/main/java/com/UI/InfoAdapter.java
package com.UI;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.cerqueira.mellina.testeandroidsantander.R;

import java.util.List;

public class InfoAdapter extends RecyclerView.Adapter<InfoAdapter.ItemHolder>  {

    private List<Info> Infos;
    private Context context;

    public InfoAdapter(Context context, List<Info> Infos) {
        this.context = context;
            this.Infos = Infos;

    }

    @Override
    public ItemHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.adapter_info, parent, false);
        ItemHolder holder = new ItemHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(ItemHolder holder, int position) {
        Info info = Infos.get(position);

        holder.txtName.setText(String.valueOf(info.getName()));
        holder.txtData.setText(info .getData());


    }

    @Override
    public int getItemCount() {
        return Infos.size();
    }

    public class ItemHolder extends RecyclerView.ViewHolder {
        public TextView txtName;
        public TextView txtData;



        public ItemHolder(View view) {
            super(view);
            txtName = view.findViewById(R.id.txtName);
            txtData = view.findViewById(R.id.txtData);


        }

    }
}

File: app/src/main/java/com/galdino/testandroid/plataform/views/investment/InfoAdapter.kt
package com.galdino.testandroid.plataform.views.investment

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.galdino.testandroid.R
import com.galdino.testandroid.data.entity.investment.DownInfo
import com.galdino.testandroid.data.entity.investment.InfoModel
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.adapter_down_info.*
import kotlinx.android.synthetic.main.adapter_info.*

class InfoAdapter(private val mList: List<InfoModel>,
                  private val isDown: Boolean = false,
                  private val mListener: InfoAdapter.Listener? = null): RecyclerView.Adapter<InfoAdapter.ViewHolder>() {
    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(viewGroup.context)
        val inflate = if(isDown){
            inflater.inflate(R.layout.adapter_down_info, viewGroup, false)
        }
        else{
            inflater.inflate(R.layout.adapter_info, viewGroup, false)
        }

        return ViewHolder(inflate)
    }

    override fun getItemCount(): Int {
        return mList.size
    }

    interface Listener{
        fun onDownloadClicked(downInfo: DownInfo)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val infoModel = mList[position]
        if(isDown)
        {
            holder.tvNameDown.text = infoModel.name
            holder.vDownload.setOnClickListener {
                mListener?.onDownloadClicked(mList[position] as DownInfo)
            }
        }
        else{
            holder.tvName.text = infoModel.name
            holder.tvData.text = infoModel.data
        }
    }

    class ViewHolder(override val containerView: View) : LayoutContainer, RecyclerView.ViewHolder(containerView) {

    }
}